sematic/resolvers/state_machine_resolver.py

```diff
@@ -50,10 +50,23 @@ def resolve(self, future: AbstractFuture) -> typing.Any:
                 for future_ in self._futures:
                     if future_.state == FutureState.CREATED:
                         self._schedule_future_if_args_resolved(future_)
+                        continue
                     if future_.state == FutureState.RETRYING:
                         self._execute_future(future_)
+                        continue
                     if future_.state == FutureState.RAN:
                         self._resolve_nested_future(future_)
+                        continue
```

Comment:
These changes are related to that other issue we discussed, about things not getting scheduled as soon as possible in certain parallel situations. I'm fine with you keeping it in if you'd like, but we may want to test it in that case using some setup like the one in the error-case I described.

Reply:
If you're requesting proof of correctness, does [this](https://dev.sematic.cloud/pipelines/sematic.examples.add.pipeline.pipeline/d4126e57611e49508754b3a3abb3878f) cover it? (It's almost identical to the use case you described)
If not, could you please clarify what to validate?

![image](https://user-images.githubusercontent.com/1894533/196306851-772cf709-b82b-4e59-b9cb-437a71321a3d.png)